miniapps/dpg/util/pml.hpp

```diff
@@ -0,0 +1,133 @@
+// Copyright (c) 2010-2022, Lawrence Livermore National Security, LLC. Produced
+// at the Lawrence Livermore National Laboratory. All Rights reserved. See files
+// LICENSE and NOTICE for details. LLNL-CODE-806117.
+//
+// This file is part of the MFEM library. For more information and source code
+// availability visit https://mfem.org.
+//
+// MFEM is free software; you can redistribute it and/or modify it under the
+// terms of the BSD-3 license. We welcome feedback and contributions, see file
+// CONTRIBUTING.md for details.
+
+#include "mfem.hpp"
+
+namespace mfem
+{
+
+/// Class for setting up a simple Cartesian PML region
+class CartesianPML
+{
+private:
+   Mesh *mesh;
+
+   /// Length of the PML Region in each direction
+   Array2D<double> length;
+
+   /// Computational Domain Boundary
+   Array2D<double> comp_dom_bdr;
+
+   /// Domain Boundary
+   Array2D<double> dom_bdr;
+
+   /** Integer Array identifying elements in the pml
+       0: in the pml, 1: not in the pml */
+   Array<int> elems;
+
+   /// Compute Domain and Computational Domain Boundaries
+   void SetBoundaries();
+
+public:
+   CartesianPML(Mesh *mesh_,Array2D<double> length_);
```

Comment:
Should document the `length_` array. I'm guessing this is of size `dim x 2` but it could also be `2 x dim` or my guess could be completely wrong...